splinterd/src/daemon.rs

```diff
@@ -699,29 +693,30 @@ fn start_health_service(
 }
 
 #[cfg(feature = "biome")]
-fn build_biome_routes(db_url: &str) -> Result<BiomeRestResourceManager, StartError> {
+fn build_biome_routes(db_url: String) -> Result<BiomeRestResourceManager, StartError> {
     info!("Adding biome routes");
-    let connection_pool: ConnectionPool =
-        database::ConnectionPool::new_pg(db_url).map_err(|err| {
-            StartError::RestApiError(format!(
-                "Unable to connect to the Splinter database: {}",
-                err
-            ))
-        })?;
+    let storage_location = if db_url.starts_with("postgres://") {
```

Comment:
Not sure I see the value of exposing this here, or at least, having a `StorageLocation::from_str` function.  That should return an error if it's an unsupported value. Both the `from_str`function and the create_store_factory can have feature guards for the available store backends

Reply:
Added a string parsing method (implementation of the `FromStr` trait).